plugin.video.youtube/resources/lib/youtube_plugin/kodion/service.py

```diff
@@ -0,0 +1,69 @@
+__author__ = 'bromix'
+
+from datetime import datetime
+import time
+
+from .impl import Context
+from .utils import YouTubeMonitor, YouTubePlayer
+
+
+def strptime(stamp, stamp_fmt):
+    import _strptime
+    try:
+        time.strptime('01 01 2012', '%d %m %Y')  # dummy call
+    except:
+        pass
+    return time.strptime(stamp, stamp_fmt)
+
+
+def get_stamp_diff(current_stamp):
+    stamp_format = '%Y-%m-%d %H:%M:%S.%f'
+    current_datetime = datetime.now()
+    if not current_stamp: return 86400  # 24 hrs
+    try:
+        stamp_datetime = datetime(*(strptime(current_stamp, stamp_format)[0:6]))
+    except ValueError:  # current_stamp has no microseconds
+        stamp_format = '%Y-%m-%d %H:%M:%S'
+        stamp_datetime = datetime(*(strptime(current_stamp, stamp_format)[0:6]))
+
+    time_delta = current_datetime - stamp_datetime
+    total_seconds = 0
+    if time_delta:
+        total_seconds = ((time_delta.seconds + time_delta.days * 24 * 3600) * 10 ** 6) // (10 ** 6)
+    return total_seconds
+
+
+def run():
+    sleep_time = 10
+    ping_delay_time = 60
+    ping_timestamp = None
+    first_run = True
+
+    context = Context(plugin_id='plugin.video.youtube')
+
+    context.log_debug('YouTube service initialization...')
+    version = context.get_system_version().get_version()
+    settings = context.get_settings()
+
+    player = YouTubePlayer(context=context)
+    monitor = YouTubeMonitor()
+
+    monitor.remove_temp_dir()
+
+    while not monitor.abortRequested():
```

Comment:
I don't think this works across all versions but is fine according to the branch you are submitting. Probably the checks you have above for the listitem art stuff for legacy versions of kodi can be removed

Reply:
added in v14 so should be fine in 15.